hack/istio/tempo/install-tempo-env.sh

```diff
@@ -0,0 +1,160 @@
+      shift;shift
+      ;;
+    -ib|--install-bookinfo)
+      INSTALL_BOOKINFO="$2"
+      shift;shift
+      ;;
+    -ii|--install-istio)
+      INSTALL_ISTIO="$2"
+      shift;shift
+      ;;
+    -t|--tempo-ns)
+      TEMPO_NS="$2"
+      shift;shift
+      ;;
+    -h|--help)
+      cat <<HELPMSG
+Valid command line arguments:
+  -c|--client:
+       client exe. Just kubectl is supported at the moment.
+  -da|--delete-all:
+       Delete tempo and all the components installed (Including Istio, Kiali & bookinfo).
+  -dt|--delete-tempo:
+       Delete tempo, tempo operator and cert manager.
+  -ib|--install-bookinfo:
+       If bookinfo should be installed. true by default.
+  -ii|--install-istio:
+       If istio should be installed. true by default.
+  -t|--tempo-ns:
+       Tempo namespace. Tempo by default.
+  -h|--help:
+       this message
+HELPMSG
+      exit 1
+      ;;
+    *)
+      echo "ERROR: Unknown argument [$key]. Aborting."
+      exit 1
+      ;;
+  esac
+done
+
+SCRIPT_DIR="$(dirname "${BASH_SOURCE[0]}")"
+
+if [ "${DELETE_ALL}" == "true" ]; then
+  DELETE_TEMPO="true"
+fi
+
+if [ "${DELETE_TEMPO}" == "true" ]; then
+  echo -e "Deleting tempo \n"
+  ${CLIENT_EXE} delete -f https://github.com/cert-manager/cert-manager/releases/latest/download/cert-manager.yaml
+  ${CLIENT_EXE} delete -f https://github.com/grafana/tempo-operator/releases/latest/download/tempo-operator.yaml
+  ${CLIENT_EXE} delete secret -n ${TEMPO_NS} tempostack-dev-minio
+  ${CLIENT_EXE} delete TempoStack smm -n ${TEMPO_NS}
+  ${CLIENT_EXE} delete ns ${TEMPO_NS}
+  if [ "${DELETE_ALL}" == "true" ]; then
+    ${SCRIPT_DIR}/../install-istio-via-istioctl.sh -c ${CLIENT_EXE} -di true
+    ${SCRIPT_DIR}/../install-bookinfo-demo.sh -c ${CLIENT_EXE} -db true
+  fi
+else
+  echo -e "Installing cert manager...\n"
+  ${CLIENT_EXE} apply -f https://github.com/cert-manager/cert-manager/releases/latest/download/cert-manager.yaml
+  echo -e "Waiting for cert-manager pods to be ready... \n"
+  $CLIENT_EXE wait pods --all -n cert-manager --for=condition=Ready --timeout=5m
+
+  echo -e "Installing latest Tempo operator \n"
+  ${CLIENT_EXE} apply -f https://github.com/grafana/tempo-operator/releases/latest/download/tempo-operator.yaml
+  echo -e "Waiting for Tempo operator to be ready... \n"
+  $CLIENT_EXE wait pods --all -n tempo-operator-system --for=condition=Ready --timeout=5m
+
+  ${CLIENT_EXE} create namespace ${TEMPO_NS}
+
+  echo -e "Installing minio and create secret \n"
+  ${CLIENT_EXE} apply --namespace ${TEMPO_NS} -f ${SCRIPT_DIR}/minio.yaml
+
+  ${CLIENT_EXE} create secret generic -n ${TEMPO_NS} tempostack-dev-minio \
+    --from-literal=bucket="tempo-data" \
+    --from-literal=endpoint="http://minio:9000" \
+    --from-literal=access_key_id="minio" \
+    --from-literal=access_key_secret="minio123"
+
+  echo -e "Installing tempo \n"
+  ${CLIENT_EXE} apply -n ${TEMPO_NS} -f - <<EOF
+apiVersion: tempo.grafana.com/v1alpha1
+kind: TempoStack
+metadata:
+  name: smm
```

Comment:
I'm just curious - what is "smm" an abbreviation for?

Reply:
Ah, probably that was from a template. Does cr make more sense?